src/TorchSharp/TorchVision/models/VGG.cs

```diff
@@ -365,24 +387,40 @@ public VGG(string name, int numClasses, bool batch_norm, float dropout = 0.5f, D
 
                 RegisterComponents();
 
-                foreach (var (_, m) in named_modules()) {
-                    switch (m) {
-                    // This test must come before the Tensor test
-                    case TorchSharp.Modules.Conv2d conv:
-                        torch.nn.init.kaiming_normal_(conv.weight, mode: init.FanInOut.FanOut, nonlinearity: init.NonlinearityType.ReLU);
-                        if (conv.bias is not null && !conv.bias.IsInvalid) {
-                            torch.nn.init.constant_(conv.bias, 0);
+                if (string.IsNullOrEmpty(weights_file)) {
+
+                    foreach (var (_, m) in named_modules()) {
+                        switch (m) {
+                        // This test must come before the Tensor test
+                        case TorchSharp.Modules.Conv2d conv:
+                            torch.nn.init.kaiming_normal_(conv.weight, mode: init.FanInOut.FanOut, nonlinearity: init.NonlinearityType.ReLU);
+                            if (conv.bias is not null && !conv.bias.IsInvalid) {
+                                torch.nn.init.constant_(conv.bias, 0);
+                            }
+                            break;
+                        case TorchSharp.Modules.BatchNorm2d bn:
+                            torch.nn.init.constant_(bn.weight, 1);
+                            torch.nn.init.constant_(bn.bias, 0);
+                            break;
+                        case TorchSharp.Modules.Linear ln:
+                            torch.nn.init.normal_(ln.weight, 0, 0.01);
+                            torch.nn.init.constant_(ln.bias, 0);
+                            break;
                         }
-                        break;
-                    case TorchSharp.Modules.BatchNorm2d bn:
-                        torch.nn.init.constant_(bn.weight, 1);
-                        torch.nn.init.constant_(bn.bias, 0);
-                        break;
-                    case TorchSharp.Modules.Linear ln:
-                        torch.nn.init.normal_(ln.weight, 0, 0.01);
-                        torch.nn.init.constant_(ln.bias, 0);
-                        break;
                     }
+                } else {
+
+                    foreach (var (_, m) in named_modules()) {
+                        switch (m) {
```

Comment:
>switch (m) { [](http://example.com/codeflow?start=24&length=12)

Is it expected to add more cases to this switch statement. I am asking if we can have this if statement instead?